platform/api.scripting/test/unit/src/org/netbeans/api/scripting/JavaScriptEnginesTest.java

```diff
@@ -256,7 +256,7 @@ public void allowLoadAClassInJS() throws Exception {
         // BEGIN: org.netbeans.api.scripting.JavaScriptEnginesTest#allowLoadAClassInJS
         Object fn = engine.eval("(function(obj) {\n"
                 + "  var Long = Java.type('java.lang.Long');\n"
-                + "  return new Long(33);\n"
+                + "  return new Long(\"33\");\n"
```

Comment:
Is this change really necessary? It does work in the GraalVM 20.2.0 console (CCing @woess):
```bash
./graalvm-ce-java8-20.2.0/bin/js --jvm
> let Long = Java.type("java.lang.Long");
> new Long(33)
33
> new Long("33")
33
```
It would be surprising that the interop behaves differently in the console and in our `ScriptEngine`...

Reply:
Well, [this is the error](https://travis-ci.org/github/apache/netbeans/jobs/727721920#L8773) that I ran into in CI and also later confirmed locally. Maybe there is a way to turn this off `ScriptEngine` configuration?

Reply:
aa76809 seems to be a workaround. Requesting Nashorn compatibility mode activates [NASHORN_HOST_ACCESS](https://github.com/graalvm/graaljs/blob/vm-20.2.0/graal-js/src/com.oracle.truffle.js.scriptengine/src/com/oracle/truffle/js/scriptengine/GraalJSScriptEngine.java#L270) which contains the Long ⇒ String [conversion](https://github.com/graalvm/graaljs/commit/c09ae303ee6b7d9a1a470cade79b147c15ea6d46) as identified by Samuel. By calling
```java
b.put("polyglot.js.allowHostAccess", true);
```
after that one replaces the Nashorn compatibility host access with `HostAccess.ALL` switching back to the behavior of the engine as of GraalVM 19.3.x. CCing @iamstolis

Reply:
Yes, this is a known consequence of the new conversion and also a known workaround that removes the new conversion see https://github.com/graalvm/graaljs/issues/286

Note that the conversion itself is not incorrect (in the sense that Nashorn has it as well and so some Nashorn users need it). Unfortunately, the inclusion of the conversion discloses that our method resolution (in `truffle`) is different than Nashorn one.